spec/ddtrace/workers/async_spec.rb

```diff
@@ -0,0 +1,427 @@
+require 'spec_helper'
+
+require 'ddtrace/worker'
+require 'ddtrace/workers/async'
+
+RSpec.describe Datadog::Workers::Async::Thread do
+  context 'when included into a worker' do
+    subject(:worker) { worker_class.new(&task) }
+
+    let(:worker_class) do
+      Class.new(Datadog::Worker) { include Datadog::Workers::Async::Thread }
+    end
+
+    let(:task) { proc { |*args| worker_spy.perform(*args) } }
+    let(:worker_spy) { double('worker spy') }
+
+    before { allow(worker_spy).to receive(:perform) }
+
+    shared_context 'perform and wait' do
+      let(:perform) { worker.perform(*args) }
+      let(:args) { [:foo, :bar] }
+      let(:perform_complete) { ConditionVariable.new }
+      let(:perform_result) { double('perform result') }
+      let(:perform_task) { proc { |*_args| } }
+
+      before do
+        allow(worker_spy).to receive(:perform) do |*actual_args|
+          perform_task.call(*actual_args)
+          perform_complete.signal
+          perform_result
+        end
+
+        perform
+
+        # Block until #perform gives signal or timeout is reached.
+        Mutex.new.tap do |mutex|
+          mutex.synchronize do
+            perform_complete.wait(mutex, 0.1)
+            sleep(0.1) # Give a little extra time to collect the thread.
+          end
+        end
+      end
+    end
+
+    shared_context 'perform and wait with error' do
+      include_context 'perform and wait' do
+        let(:error) { error_class.new }
+        let(:error_class) { stub_const('TestError', Class.new(StandardError)) }
+        let(:perform_task) do
+          proc do |*_actual_args|
+            raise error
+          end
+        end
+      end
+    end
+
+    describe '#perform' do
+      subject(:perform) { worker.perform(*args) }
+      let(:args) { [:foo, :bar] }
+
+      context 'given arguments' do
+        include_context 'perform and wait' do
+          let(:perform_task) do
+            proc do |*actual_args|
+              expect(actual_args).to eq args
+            end
+          end
+        end
+
+        it 'performs the task async' do
+          # expect(worker.result).to eq(perform_result)
```

Comment:
I think we forgot to clean up this line.

Reply:
Mmmm, yes, should fix that.